packages/wrapped-keys/src/lib/types.ts

```diff
@@ -0,0 +1,170 @@
+ *
+ * @typedef ExportPrivateKeyParams
+ * @extends BaseApiParams
+ *
+ */
+export type ExportPrivateKeyParams = BaseApiParams;
+
+/** Includes the decrypted private key and metadata that was stored alongside it in the wrapped keys service
+ *
+ * @typedef ExportPrivateKeyResult
+ * @property { LIT_NETWORKS_KEYS } litNetwork The LIT network that the LIT Node Client was configured for when the key was persisted to the wrapped keys service
+ * @property { string } decryptedPrivateKey The decrypted, plain text private key that was persisted to the wrapped keys service
+ * @property { string } pkpAddress The LIT PKP Address that the key was linked to; this is derived from the provided pkpSessionSigs
+ * @property { string } publicKey The public key of the key being imported into the wrapped keys service
+ * @property { string } keyType The algorithm type of the key; this might be K256, ed25519, or other key formats.  The `keyType` will be included in the metadata returned from the wrapped keys service
+ *
+ */
+export interface ExportPrivateKeyResult {
+  pkpAddress: string;
+  decryptedPrivateKey: string;
+  publicKey: string;
+  litNetwork: LIT_NETWORKS_KEYS;
+  keyType: string;
+}
+
+/** @typedef GeneratePrivateKeyParams
+ * @extends BaseApiParams
+ * @property {Network} network The network for which the private key needs to be generated; keys are generated differently for different networks
+ */
+export type GeneratePrivateKeyParams = BaseApiParams &
+  ApiParamsSupportedNetworks;
+
+/** @typedef GeneratePrivateKeyResult
+ * @property { string } pkpAddress The LIT PKP Address that the key was linked to; this is derived from the provided pkpSessionSigs
+ * @property { string } generatedPublicKey The public key component of the newly generated keypair
+ *
+ */
+export interface GeneratePrivateKeyResult {
+  pkpAddress: string;
+  generatedPublicKey: string;
+}
+
+/** @typedef ImportPrivateKeyParams
+ * @extends BaseApiParams
+ *
+ * @property { string } privateKey The private key to be imported into the wrapped keys service
+ * @property { string } publicKey The public key of the key being imported into the wrapped keys service
+ * @property { string } keyType The algorithm type of the key; this might be K256, ed25519, or other key formats.  The `keyType` will be included in the metadata returned from the wrapped keys service
+ */
+export interface ImportPrivateKeyParams extends BaseApiParams {
+  privateKey: string;
+  publicKey: string;
+  keyType: string;
+}
+
+interface SignMessageParams {
+  messageToSign: string | Uint8Array;
+}
+
+/** @typedef SignMessageWithEncryptedKeyParams
+ * @extends BaseApiParams
+ *
+ * @property { string | Uint8Array } messageToSign The message to be signed
+ */
+export type SignMessageWithEncryptedKeyParams = BaseApiParams &
+  ApiParamsSupportedNetworks &
+  SignMessageParams;
+
+interface BaseLitTransaction {
+  chain: string;
```

Comment:
It would be nice if we could restrict this to the [supported EVM networks](https://github.com/LIT-Protocol/js-sdk/blob/cc95304922d0e0b00300e0198de2455586b858a4/packages/constants/src/lib/constants/constants.ts#L29-L601) for `EthereumLitTransaction`s and `mainnet-beta | testnet | devnet` for Solana txs

Reply:
Sure for later, can you cut a ticket for tracking this @spacesailor24 and comment here